overlord/snapstate/aliases.go

```diff
@@ -187,11 +187,13 @@ func (m *SnapManager) doAlias(t *state.Task, _ *tomb.Tomb) error {
 		}
 		switch newStatus {
 		case "enabled":
-			err := checkAliasConflict(st, snapName, alias)
-			if err != nil {
-				return err
+			if aliasStatuses[alias] != "auto" {
+				err := checkAliasConflict(st, snapName, alias)
```

Comment:
I don't understand the comment here, we don't check for conflict if it's auto because then the alias is already there we are just changing its status from "auto" to explicitly "enabled"